tree/20_cfe_basics/log_rudder.cf

```diff
@@ -29,212 +29,17 @@
-      # We put those files into the state directory to avoid purging them during each update
-      # (which would break the current run's reporting).
-      # We add the PID to the file name to be able to handle cuncurrent agent runs with separate expected reports
-      "expected_reports_temp"   string => "${expected_reports_destination}.tmp";
-      "expected_reports_file"   string => "${expected_reports_destination}.res";
-
-    logger_rudder_final_resfile_repaired::
-      "c_class_prefix" string => canonify("${old_class_prefix}");
-
-      # 4/ Once the final expected reports file has been expanded, read in our array
-      "number_lines"
-        int        => getfields("^${current_technique_report_info.technique_name};;${c_class_prefix};;.*", "${expected_reports_file}", ";;", "report_data");
-
-  classes:
-    logger_rudder_final_resfile_repaired::
-      "report_data_read"    expression => isgreaterthan("${number_lines}","0");
-  
-      # We want to detect if expected report data is available in v2 format.
-      # V2 format is characterized by a new field for a total of 6 fields.
-      # The file is rewritten first in .tmp file then in .res file
-      # The second rewrite is done using the edit_line below, so if the 6th field doesn't exist in the original file
-      # then it is translater to a litteral "$reports[0][5]}" in the file.
-      # If there has been no rewrite at all, then there is no 6th field at all in the final file
-
-      # 1/ The file has been rewritten with an empty field
-      # 5 vs 6 : because getfields and readstringarrayidx do not have the same indexing method (getfields starts at 1)
-      "expected_report_v2_csv"     not => strcmp("${report_data[6]}","${const.dollar}{reports[0][5]}");
-
-      # 2/ The file has never been rewritten
-      "expected_report_v2_res"     not => strcmp("${report_data[6]}", "${const.dollar}{report_data[6]}");
-
-      # 3/ We add another condition : the method promiser must have been rewritten in the technique
-      #    The previous version of ncf always put "method_call" as a promiser, so this indicates an old technique version
-      "method_v2"                  not => strcmp("method_call", nth("this.callers_promisers", "1"));
-
-      # 4/ We check that the new class_prefix is not empty, otherwise the v2 method would break
-      "class_prefix_v2"            not => strcmp("${origin_class_prefix}", "");
-
-      # 5/ we aggregate this with the availability of the promiser stack to know if we should report using v2 format
-      "report_v2"           expression => "expected_report_v2_csv.expected_report_v2_res.method_v2.has_promiser_stack.class_prefix_v2";
-
-    any::
-      "no_old_class_prefix" expression => strcmp("${old_class_prefix}", "");
-
-      "pass3" expression => "pass2";
-      "pass2" expression => "pass1";
-      "pass1" expression => "any";
+    "c_class_prefix" string => canonify("${old_class_prefix}");
```

Comment:
could you add 2 spaces of indentation ?

Reply:
It's correctly idented, there is no class defined

<pre>
  vars:
    "c_class_prefix" ...
</pre>

Reply:
or do indented - spacedces even iff there is no class ?

Reply:
yes, even if no class

Reply:
OK from what i read elsewhere we do likewise !! thanks @ncharles  !

Reply:
ok i reindented with 6 spaced (and the report below is also now indented with 6 spaces)